### PR TITLE
Add SMS send and receive support

### PR DESCRIPTION
**Pull request by sandeepmistry**
Ported from MKRGSM.

cc/ @8bitkick

### CODE REVIEW

File: examples/ReceiveSMS/ReceiveSMS.ino [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+#include "arduino_secrets.h" 
+// Please enter your sensitive data in the Secret tab or arduino_secrets.h
+// PIN Number
+const char PINNUMBER[] = SECRET_PINNUMBER;
+
+// initialize the library instances
+NB nbAccess;
+NB_SMS sms;
+
+// Array to hold the number a SMS is retreived from
```
Comment (per1234) [inline examples/ReceiveSMS/ReceiveSMS.ino]:
```suggestion
// Array to hold the number an SMS is retrieved from
```

File: examples/SendSMS/SendSMS.ino [hunk 2]
```diff
@@ -0,0 +1,100 @@
+/*
+ SMS sender
+
+ This sketch, for the MKR NB 1500 board,sends an SMS message
```
Comment (per1234) [inline examples/SendSMS/SendSMS.ino]:
```suggestion
 This sketch, for the MKR NB 1500 board, sends an SMS message
```

File: examples/SendSMS/SendSMS.ino [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    }
+  }
+
+  Serial.println("NB initialized");
+}
+
+void loop() {
+
+  Serial.print("Enter a mobile number: ");
+  char remoteNum[20];  // telephone number to send sms
```
Comment (per1234) [inline examples/SendSMS/SendSMS.ino]:
```suggestion
  char remoteNum[20];  // telephone number to send SMS
```

File: examples/SendSMS/SendSMS.ino [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  readSerial(remoteNum);
+  Serial.println(remoteNum);
+
+  // sms text
```
Comment (per1234) [inline examples/SendSMS/SendSMS.ino]:
```suggestion
  // SMS text
```

**Comment by Rocketct** [comment]
i have just reviewed the code but not tested because i don't have an available SIM to test the Tx and Rx of the SMS

**Comment by sandeepmistry** [comment]
@per1234 thanks for the suggestions!

**Comment by sandeepmistry** [comment]
@SimonePDA the reference pages need to be updated for this.

**Comment by SimonePDA** [comment]
Have a look here
https://edit.arduino.cc/en/Reference/MKRNB

confirm the added APIs

* [[MKRNB_SMS_Constructor | NB_SMS]] 
* [[MKRNB_SMS_beginSMS | beginSMS()]]
* [[MKRNB_SMS_ready | ready()]]
* [[MKRNB_SMS_endSMS | endSMS()]]
* [[MKRNB_SMS_available | available()]]
* [[MKRNB_SMS_remoteNumber | remoteNumber()]]
* [[MKRNB_SMS_read | read()]]
* [[MKRNB_SMS_write | write()]]
* [[MKRNB_SMS_print | print()]]
* [[MKRNB_SMS_peek | peek()]]
* [[MKRNB_SMS_flush | flush()]]

and give OK for push

simone

**Comment by fargit** [comment]
Hello 
I am using your code but I modified my device to receive the messages with the internet.
I would like to ask you, how cam I add a part to your code enable me to send a specific code like "$65!"
to direct me for power on and off the device or just simply the rest.
As I closed my direct AT communication with the device so I am trying to add that here in peek but not successful yet. 
I appreciate  your help.

**Pull request merged**